tools/pathkit/dart/lib/pathkit.dart

```diff
@@ -0,0 +1,381 @@
+          proxy.close();
+          break;
+      }
+    }
+    assert(index == points.length);
+  }
+
+  /// The list of path verbs in this path.
+  ///
+  /// This may not match the verbs supplied by calls to [moveTo], [lineTo],
+  /// [cubicTo], and [close] after [applyOp] is invoked.
+  ///
+  /// This list determines the meaning of the [points] array.
+  Iterable<PathVerb> get verbs {
+    _updatePathData();
+    final int count = _pathData!.ref.verb_count;
+    return List<PathVerb>.generate(count, (int index) {
+      return PathVerb.values[_pathData!.ref.verbs.elementAt(index).value];
+    }, growable: false);
+  }
+
+  /// The list of points to use with [verbs].
+  ///
+  /// Each verb uses a specific number of points, specified by the
+  /// [pointsPerVerb] map.
+  Float32List get points {
+    _updatePathData();
+    return _pathData!.ref.points.asTypedList(_pathData!.ref.point_count);
+  }
+
+  void _updatePathData() {
+    assert(_path != null);
+    _pathData ??= _dataFn(_path!);
+  }
+
+  void _resetPathData() {
+    if (_pathData != null) {
+      _destroyDataFn(_pathData!);
+    }
+    _pathData = null;
+  }
+
+  /// Adds a move verb to the absolute coordinates x,y.
+  void moveTo(double x, double y) {
+    assert(_path != null);
+    _resetPathData();
+    _moveToFn(_path!, x, y);
+  }
+
+  /// Adds a line verb to the absolute coordinates x,y.
+  void lineTo(double x, double y) {
+    assert(_path != null);
+    _resetPathData();
+    _lineToFn(_path!, x, y);
+  }
+
+  /// Adds a cubic Bezier curve with x1,y1 as the first control point, x2,y2 as
+  /// the second control point, and end point x3,y3.
+  void cubicTo(
+    double x1,
+    double y1,
+    double x2,
+    double y2,
+    double x3,
+    double y3,
+  ) {
+    assert(_path != null);
+    _resetPathData();
+    _cubicToFn(_path!, x1, y1, x2, y2, x3, y3);
+  }
+
+  /// Adds a close command to the start of the current contour.
+  void close() {
+    assert(_path != null);
+    _resetPathData();
+    _closeFn(_path!, true);
+  }
+
+  void reset() {
+    assert(_path != null);
+    _resetPathData();
+    _resetFn(_path!);
+  }
+
+  /// Releases native resources.
+  ///
+  /// After calling dispose, this class must not be used again.
+  void dispose() {
+    assert(_path != null);
+    _resetPathData();
+    _destroyFn(_path!);
+    _path = null;
+  }
+
+  void applyOp(Path other, PathOp op) {
+    assert(_path != null);
+    assert(other._path != null);
+    _resetPathData();
+    _opFn(_path!, other._path!, op.index);
+  }
+}
+
+// TODO(dnfield): Figure out where to put this.
+// https://github.com/flutter/flutter/issues/99563
+final ffi.DynamicLibrary _dylib = () {
+  if (Platform.isWindows) {
+    return ffi.DynamicLibrary.open('pathkit.dll');
+  } else if (Platform.isIOS || Platform.isMacOS) {
+    return ffi.DynamicLibrary.open('libpathkit.dylib');
+  } else if (Platform.isAndroid || Platform.isLinux) {
+    return ffi.DynamicLibrary.open('libpathkit.so');
```

Comment:
Can we use wasm to avoid having to ship platform-specific libraries? I'm not sure what the wasm support level is like in the Dart VM.

Reply:
I made Dan promise he wouldn't

Reply:
(Not really)

Reply:
In theory yes, but we'd need to ship a wasm runtime to use it and it probably won't perform quite as well.

Reply:
(Skia already has a WASM version of this that they publish called "PathKit" on NPM)